src/promptflow-evals/promptflow/evals/evaluate/_evaluate.py

```diff
@@ -58,13 +68,59 @@ def _validation(target, data, evaluators, output_path, tracking_uri, evaluation_
         if not isinstance(evaluation_name, str):
             raise ValueError("evaluation_name must be a string.")
 
+    _validate_columns(data, evaluators, target)
+
+
+def _validate_columns(data, evaluators, target):
     try:
         data_df = pd.read_json(data, lines=True)
     except Exception as e:
-        raise ValueError(f"Failed to load data from {data}. Please validate it is a valid jsonl data. Error: {str(e)}.")
+        raise ValueError(
+            f"Failed to load data from {data}. Please validate it is a valid jsonl data. Error: {str(e)}.")
 
-    for evaluator_name, evaluator in evaluators.items():
-        _validate_input_data_for_evaluator(evaluator, evaluator_name, data_df)
+    if target:
+        # If the target function is given, it may return
+        # several columns and hence we cannot check the availability of columns
+        # without knowing target function semantics.
+        # Instead, here we will validate the columns, taken by target.
+        _validate_input_data_for_evaluator(target, None, data_df, is_target_fn=True)
+    else:
+        for evaluator_name, evaluator in evaluators.items():
+            _validate_input_data_for_evaluator(evaluator, evaluator_name, data_df)
+
+
+def _apply_target_to_data(target: Callable, data: str, pf_client: PFClient) -> str:

```

Comment:
I think this might not work because we don't want target function to be called for each evaluator. Target function should be only called once.

Reply:
Yes, I did the research, it will actually fail because we have [two ways](https://github.com/microsoft/promptflow/blob/d614febf089ad05cdecc79163f6b4fc6e68fc20d/src/promptflow-evals/promptflow/evals/evaluate/_evaluate.py#L108) to execute code: using PromptFlow (the approach above will work) and, if the evaluator is not functional class, we will use CodeClient, which [does not](https://github.com/microsoft/promptflow/blob/d614febf089ad05cdecc79163f6b4fc6e68fc20d/src/promptflow-evals/promptflow/evals/evaluate/_code_client/code_client.py#L63C5-L63C83) take into account previous runs.